code/model/DMSDocument.php

```diff
@@ -67,53 +91,69 @@ class DMSDocument extends DataObject implements DMSDocumentInterface
      */
     private static $default_download_behaviour = 'download';
 
-    /**
-     * @param Member $member
-     *
-     * @return boolean
-     */
     public function canView($member = null)
     {
         if (!$member || !(is_a($member, 'Member')) || is_numeric($member)) {
             $member = Member::currentUser();
         }
 
-        // extended access checks
-        $results = $this->extend('canView', $member);
-
-        if ($results && is_array($results)) {
-            if (!min($results)) {
-                return false;
-            }
```

Comment:
These extension checks are important - can you add them back in? Without them the checks aren't extensible from user code